src/LearnMe.Infrastructure/Repository/CalendarEventsRepository.cs

```diff
@@ -18,13 +18,23 @@ public CalendarEventsRepository(ApplicationDbContext context) : base(context)
             _context = context;
         }
 
+        public async Task<bool> DeleteByCalendarIdAsync(string calendarId)
+        {
+            var eventToBeDeleted = await GetByCalendarIdAsync(calendarId);
+
+            return await DeleteAsync(eventToBeDeleted.Id);

```

Comment:
Dlaczego nie usuniesz od razu `eventToBeDeleted`? Nie musisz  jej szukać ponownie. EF ją trackuje. 

Dobra, widzę, że chcesz by result z `GetByCalendarIdAsync` był no tracking, jest jakiś powód dlaczego chcesz tak robić? Bo gdyby nie `detached` z metody wyzej mogłabyś usunąć encje. Przy `DeleteAsync` masz dodatkowy strzał na bazy , po co? 50-200ms w plecy (zależy od prędkości łącza)

Reply:
`GetByCalendarIdAsync` jest używane w Calendar.cs i LessonRepository.cs, także wolałam to oddzielić od innych zapytań, które później będę robić na bazie, a do których potrzebuje najpierw wydarzenia z takim CalendarId.

Reply:
A może ja to powinnam od razu w zapytaniu napisać... muszę to przemyśleć

Reply:
![obraz](https://user-images.githubusercontent.com/19174792/99908601-dce31500-2ce3-11eb-9d16-11042839b999.png)

Reply:
Co o tym sądzisz?

Reply:
No lepiej, ale trochę bez sensu. Nie możesz przeciążyć `DeleteAsync()` by przyjmowała za parametr albo encje albo Id jeśli nie chcesz refaktorować całej apki? 

```csharp
public bool DeleteAsync(int id);
{
}

public bool DeleteAsync(T entity);
{
}
```

Reply:
A no i tak to co robisz jest źle, bo pobierasz raz encje w `DeleteByCalendarIdAsync()` by potem to samo zrobić w `DeleteAsync()`, każdy ci powie że to bezsens nawet ten co nie jest zaznajomiony z programowaniem (pobieranie dwóch tych samych rzeczy dwa raz) ;p

Reply:
mhm no rozumiem, że 2x pobieram tą samą encję. Ale tutaj case jest taki że CalendarEvent ma po pierwsze 1) swoje Id z bazy danych (primary key) i ma też dodatkową kolumnę 2) CalendarId - który jest id-kiem z kalendarza google.
I ostatecznie żeby użyć metody Delete z generycznego repo - muszę najpierw znaleźć, jakie Id w bazie odpowiada encji z podanym CalendarId. i potem już mam metodę z gerycznego repo: Delete by Id (nie Delete by CalendarId). nie mogę podać

Reply:
bez wyciągnięcia Id widzę że nie mogę użyć metody _context.Delete(entity) - jeśli ta entity nie ma podanej propercji Id
więc zostaję na razie z tym tak jak było
`DeleteByCalendarIdAsync()` jest znowu potrzebne np. przy synchronizacji przed Update i w tym samym miejscu przed Delete

Reply:
Ale te metode możesz sobie dopisać. Nie mieszaj w to żadnego `Id` albo `CalendarId`. Dostajesz Encje z bazy - usuwasz ją. Nie musisz jej szukać jeszcze raz jeśli ją dostałaś z bazy. 

https://www.learnentityframeworkcore.com/dbset/deleting-data

Reply:
To mam tą metodę w CrudRepository.cs - generycznym
`        public async Task<bool> DeleteAsync(T entity)
        {
            _context.Remove(entity);

            return await SaveAsync();
        }`

Reply:
"Dostajesz Encje z bazy - usuwasz ją. Nie musisz jej szukać jeszcze raz jeśli ją dostałaś z bazy."
czyli chodzi Ci o coś takiego:
![obraz](https://user-images.githubusercontent.com/19174792/100135094-713aac80-2e89-11eb-9d9c-c879951ebfcc.png)

Reply:
Tak, o coś takiego

Reply:
Super dzięki 👍